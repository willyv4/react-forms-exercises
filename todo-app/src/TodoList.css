.todo-list {
	
	width: 200px;
	
}