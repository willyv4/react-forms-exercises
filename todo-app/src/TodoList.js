import NewTodoForm from "./NewTodoForm";
import { useState } from "react";
import { v4 as uuid } from "uuid";
import "./TodoList.css";

const TodoList = () => {
  const [todo, setTodo] = useState([]);

  const addTodo = (todo) => {
    let newTodo = { ...todo, id: uuid() };
    setTodo((todo) => [...todo, newTodo]);
  };

  const removeTodo = (id) => {
    const currentBoxes = todo.filter((t) => t.id !== id);
    setTodo(currentBoxes);
  };

  const handleRemove = (id) => {
    removeTodo(id);
  };

  const renderTodos = () => {
    return (
      <ul>
        {todo.map((todo) => (
          <div key={todo.id}>
            <li>{todo.todo}</li>
            <button onClick={() => handleRemove(todo.id)}>X</button>
          </div>
        ))}
      </ul>
    );
  };

  return (
    <div className="todo-list">
      <NewTodoForm addTodo={addTodo} />
      {renderTodos()}
    </div>
  );
};
export default TodoList;
