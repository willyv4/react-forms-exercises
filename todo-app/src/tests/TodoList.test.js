import TodoList from "../TodoList";
import { render } from "@testing-library/react";

it("renders without crashing", function () {
  render(<TodoList />);
});

it("matches snapshot", function () {
  const { asFragment } = render(<TodoList />);
  expect(asFragment()).toMatchSnapshot();
});
