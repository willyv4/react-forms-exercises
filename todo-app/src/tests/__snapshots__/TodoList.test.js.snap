// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="todo-list"
  >
    <form>
      <label
        for="todo"
      >
        Todo:
      </label>
      <input
        id="todo"
        name="todo"
        type="text"
        value=""
      />
      <button>
        Add Todo
      </button>
    </form>
    <ul />
  </div>
</DocumentFragment>
`;
